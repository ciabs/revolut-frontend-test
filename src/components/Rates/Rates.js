import React from 'react';
import PropTypes from 'prop-types';
import FontAwesome  from 'react-fontawesome';

import {Container} from './Rates.styles';
import {getCurrencySymbol} from '../../helpers/currency.helper';

const Rates = ({rates, fromCurrency, toCurrency}) => (
  <Container>
    {
      rates.date && rates.rates[toCurrency] ? (
        <div>
          <FontAwesome
            name='line-chart'
            style={{marginRight: '5px'}}
          />
          <span>
            {getCurrencySymbol(fromCurrency)}1 = {getCurrencySymbol(toCurrency)}{rates.rates[toCurrency].toFixed(4)}
          </span>
        </div>
      ) : (
        <div>
          <FontAwesome
            name='circle-o-notch'
            spin
          />
        </div>
      )
  }
  </Container>
);


Rates.propTypes = {
  rates: PropTypes.object,
  fromCurrency: PropTypes.string,
  toCurrency: PropTypes.string
};

export default Rates;
