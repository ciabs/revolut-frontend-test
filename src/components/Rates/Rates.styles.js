import styled from 'styled-components';
import {FROM_BG_COLOR, RATE_COLOR, TO_BG_COLOR} from '../../styles/constants';

export const Container = styled.div`
  position: absolute;
  bottom: -18px;
  z-index: 2;
  background-color: ${FROM_BG_COLOR};
  border-radius: 18px;
  font-size: 14px;
  color: ${RATE_COLOR};
  left: 50%;
  transform: translateX(-50%);
  padding: 8px 12px;
  border: solid 2px ${TO_BG_COLOR};
`;

export const LoadingDot = styled.span`
  color: ${props => props.isLoading ? RATE_COLOR : FROM_BG_COLOR}
`;
