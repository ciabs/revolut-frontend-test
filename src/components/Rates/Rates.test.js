import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Rates from './Rates';

describe('Rates', () => {
  const props = {
    rates: {},
    fromCurrency: 'EUR',
    toCurrency: 'GBP',
    isLoading: false
  };

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Rates {...props} />,
      div
    );
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(
      <Rates {...props} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
