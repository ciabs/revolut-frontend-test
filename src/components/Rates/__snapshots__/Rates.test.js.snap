// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rates has a valid snapshot 1`] = `
<div
  className="sc-bdVaJa hUDFso"
>
  <div>
    <span
      aria-hidden={true}
      className="fa fa-circle-o-notch fa-spin"
    />
  </div>
</div>
`;
